apps/testclient/management/commands/create_test_users_and_applications_batch.py

```diff
@@ -187,45 +212,47 @@ def create_dev_users_apps_and_bene_crosswalks(group, bene_count, dev_count, app_
     for bpk in bene_pk_list:
         seed = randint(1, 10)
         b = User.objects.get(pk=bpk)
-        tokens = []
-        if seed <= 7:
+        # check how many apps there are because there may be only 1
+        # (caller can specify 0 for the app option)
+        if seed <= 7 or len(app_list) == 1:
             # sign up to 1 app
-            tokens.append(create_test_access_refresh_archived_tokens(b, choice(app_list)))
-        elif seed <= 9:
+            create_test_access_refresh_archived_tokens(
+                b, choice(app_list), refresh_count, archived_count, seed)
+        elif seed <= 9 and len(app_list) > 1:
             # sign up to 2 apps
             a2 = sample(app_list, 2)
```

Comment:
what does a2 represent here? The 2 sample apps created?

Reply:
It's the set of two randomly chosen apps from all the apps that exist in the database. Tokens are then generated the newly created users and each app (one token per user/app combo).

Reply:
when you say "all the apps that exist," those are sample apps in the db or actual apps?

Reply:
I mean all the apps that exist on the environment's database. This is typically run in a local Docker environment for testing (and should probable never be run on sandbox or prod environments) so your local DB shouldn't ever have "actual" apps (with real world information). Part of the Docker container coming up is creating one sample application called TestApp. The "app_list" set contains all possible applications in your environment's DB so it would include existing ones (including Test App) on top of the ones that were just created.